### DISCUSSION TITLE
Reinitialized tom select using class name

### DISCUSSION THREAD

**Discussion by vaniarizki** [orchidjs/tom-select#378] [Q&A] [answered]
I'm using tom-select forEach class name, like this:
```
document.querySelectorAll('.select').forEach((el)=>{
	let settings = {};
 	new TomSelect(el,settings);
});
```
then I want to change the value based on other option. How can I do it? I got error "Tom Select already initialized on this element". I don't know how to destroy(). Thanks

**Comment by gkaft** [comment]
Have you found any solution? 

I try this 

`let select = document.getElementById('#selectbox_id');
let control = select.tomselect;`

but this always returns null

**Reply by vaniarizki** [reply]
yes I've found the solution. 

for my case it's:
`document.querySelectorAll('.select').forEach((el)=>{
        if (el.tomselect) {
                let control = el.tomselect;
                control.destroy();
        }
	let settings = {};
 	new TomSelect(el,settings);
});`


I think if you use ID then you can use the variable directly like this:
 `let select = document.getElementById('#selectbox_id'); 
  select.clear(); `

**Comment by vaniarizki** [comment, marked as answer]
I've found the solution.

for my case it's:
`document.querySelectorAll('.select').forEach((el)=>{ 
        if (el.tomselect) { 
                  let control = el.tomselect; 
                  control.destroy(); 
       } 
       let settings = {}; 
       new TomSelect(el,settings); 
});`

**Reply by gkaft** [reply]
No this doesn't work. I have read the documentation https://tom-select.js.org/docs/api/ and tried to use 

 `let select = document.getElementById('#select');
 let control = select.tomselect;`
 
 but the select is always null...

**Reply by vaniarizki** [reply]
can you send me the full code in jsfiddle?

**Reply by gkaft** [reply]
I have put the code to jsfiddle 
https://jsfiddle.net/kaftakis/x7L25u0h/

the tomselect is not working but maybe you understand what i would like to do.

**Reply by vaniarizki** [reply]
you didn't add tomselect and jquery as resources there. this is what i understand from your code. https://jsfiddle.net/abuc8qtr/1/

**Reply by gkaft** [reply]
Thank you for your help but on my project jquery is used with webpack and tom-select objects are initialized globaly (with a foreach statement). The problem i try to solve is to change a tomselect object that is already initialized.

**Reply by vaniarizki** [reply]
have you solved your problem?

**Reply by gkaft** [reply]
No. i'm working on it

**Reply by vaniarizki** [reply]
try this one: https://jsfiddle.net/runxL4a8/

**Reply by gkaft** [reply]
Thank you very much. This worked!!!

**Reply by vaniarizki** [reply]
great!
